appserver/payara-appserver-modules/rest-monitoring/rest-monitoring-war/src/main/java/fish/payara/monitoring/rest/app/processor/SimpleTypeProcessor.java

```diff
@@ -49,11 +56,43 @@
 public class SimpleTypeProcessor implements TypeProcessor<SimpleType> {
 
     @Override
-    public Object processObject(Object object) {
-            return object;
+    public JsonValue processObject(Object object) {
+        if (SimpleType.BIGDECIMAL.isValue(object)) {
+            return Json.createValue((BigDecimal) object);
+        } else if (SimpleType.BIGINTEGER.isValue(object)) {
+            return Json.createValue((BigInteger) object);
+        } else if (SimpleType.BOOLEAN.isValue(object)) {
+            Boolean value = (Boolean) object;
+            if (value) {
+                return JsonValue.TRUE;
+            } else {
+                return JsonValue.FALSE;
+            }
+        } else if (SimpleType.DOUBLE.isValue(object)) {
+            return Json.createValue((Double) object);
+        } else if (SimpleType.INTEGER.isValue(object)) {
+            return Json.createValue((Integer) object);
+        } else if (SimpleType.LONG.isValue(object)) {
+            return Json.createValue((Long) object);
+        } else if (SimpleType.STRING.isValue(object)) {
+            return Json.createValue((String) object);
+        } else if (object == null || SimpleType.VOID.isValue(object)){
+            return JsonValue.NULL;         
+        } else if (SimpleType.BYTE.isValue(object)) {
+            return Json.createValue(((Byte) object).intValue());
+        } else if (SimpleType.CHARACTER.isValue(object)) {
+            return Json.createValue(((Character) object).toString());
+        } else if (SimpleType.DATE.isValue(object)) {
+            return Json.createValue(((Date) object).toString());
+        } else if (SimpleType.FLOAT.isValue(object)) {
+            return Json.createValue(BigDecimal.valueOf((Long) object));
+        } else if (SimpleType.OBJECTNAME.isValue(object)) {
+            return Json.createValue(((ObjectName) object).toString());
+        } else if (SimpleType.SHORT.isValue(object)) {
+            return Json.createValue(((Short) object).intValue());
+        } else {
+            throw new JsonException("Unable to process object: " + object);
+        }
     }
 
-
 }
```

Comment:
Since the body of every case is a return statement, the else is not really needed here. The following would be a bit easier on the eye:

```java
if (BIGDECIMAL.isValue(object)) {
    return createValue((BigDecimal) object);
} 

if (BIGINTEGER.isValue(object)) {
    return createValue((BigInteger) object);
}
```